### ISSUE TITLE
Add ability to rename Spreadsheets (via a new Spreadsheet.update_title)

### ISSUE THREAD

**Issue by jansim** [burnash/gspread#1011]
**Is your feature request related to a problem? Please describe.**
It's currently possible to rename worksheets, but not yet spreadsheets.

**Describe the solution you'd like**
It would be cool if there were a method `Spreadsheet.update_title` similar to `Worksheet.update_title`.

**Describe alternatives you've considered**
I thought about implementing the necessary code in a project myself, but I would need to access private properties for that, so it seems suboptimal. I would be happy to create a PR for this with a proposed solution.

**Additional context**
The google sheets API documentation has some example code for this [here](https://developers.google.com/sheets/api/guides/batchupdate) and after skimming the source code it seems like a very straightforward addition very similar to the current `Worksheet.update_title`.

API payload example:
```
{
  "requests": [{
      "updateSpreadsheetProperties": {
          "properties": {"title": "My New Title"},
          "fields": "title"
      }
  }]
}
```

**Comment by lavigne958** [comment]
Hi thank you for this proposal.

You re very welcome to open a PR to solve that issue. This feature will be useful to GSpread users.

If you don't mind could check something first please:

- after changing the title, does that change the filename in Google drive (in web browser when you see all your files in a folder).

thank you for this proposal.

**Comment by jansim** [comment]
Thanks for the input, I will take a look. Just for clarification, you mean filename here and not the file id (in the URL) right?

**Comment by lavigne958** [comment]
exactly, the name of the file when looking at a folder in Google Drive, not the file ID in the URL.

**Comment by lavigne958** [comment]
Closed by #1013
